### ISSUE TITLE
add option to add a hardcoded gradient of the action

### ISSUE THREAD

**Issue by oameye** [JuliaDynamics/CriticalTransitions.jl#29]
Performance-wise, it would be very beneficial to add the option of having a hardcoded gradient of the action function `S(x::Path)` in the (g)MAM solvers.

**Comment by reykboerner** [comment]
Yes, I guess by gradient you mean the functional derivative of `S` here with respect to the path? This would also be very useful for `langevinmcmc`. Maybe this can also be done symbolically? We have derived explicit equations for a general drift `b` and covariance matrix.

**Comment by oameye** [comment]
Yes that is exactly what I mean. In the end, it will be something in term of `b` and derivative of it.

**Comment by oameye** [comment]
In theory, this is why the Vanden-Eijnden method and sgMAM are so fast, and with these methods we can already do this. I think we can close this, as most large-deviation methods already have this capability.
